src/common/capabilities/firefox.rs

```diff
@@ -76,7 +76,15 @@ impl FirefoxCapabilities {
 
     /// Get the current list of command-line arguments to `geckodriver` as a vec.
     pub fn get_args(&self) -> Vec<String> {
-        from_value(self.capabilities["moz:firefoxOptions"]["args"].clone()).unwrap_or_default()
+        let caps = self
+            .capabilities
+            .get("moz:firefoxOptions")
+            .map(|options| options.get("args"))
+            .flatten();
+        match caps {
+            None => vec![],
+            Some(caps) => from_value(caps.clone()).unwrap_or_default(),
```

Comment:
Actually, I think this can all be done without the separate `match`.

Have a look at the chrome.rs code. I think it would be good to do this the same way.

Consider something like this:
```rust
    /// Get the specified Firefox option.
    pub fn get_firefox_option<T>(&self, key: &str) -> T
    where
        T: DeserializeOwned + Default,
    {
        self.capabilities
            .get("moz:firefoxOptions")
            .map(|options| {
                options
                    .get(key)
                    .map(|option| from_value(option.clone()).unwrap_or_default())
                    .unwrap_or_default()
            })
            .unwrap_or_default()
    }

    /// Get the current list of command-line arguments to `geckodriver` as a vec.
    pub fn get_args(&self) -> Vec<String> {
        self.get_firefox_option("args")
    }
```

Before you do, however, I'm going to see if this can be simplified further. I think we can perhaps replace the `.map()` lines with `.and_then()` and remove some nesting.

Reply:
Try this:
```rust
    /// Get the specified Firefox option.
    pub fn get_firefox_option<T>(&self, key: &str) -> T
    where
        T: DeserializeOwned + Default,
    {
        self.capabilities
            .get("moz:firefoxOptions")
            .and_then(|options| options.get(key))
            .and_then(|option| from_value(option.clone()).ok())
            .unwrap_or_default()
    }

    /// Get the current list of command-line arguments to `geckodriver` as a vec.
    pub fn get_args(&self) -> Vec<String> {
        self.get_firefox_option("args")
    }
```

Reply:
Sorry I added these comments earlier but they stayed as "pending" because I didn't realise I needed to submit the review. Let me know if this last code works. If so I'll make the same change to the `chrome.rs` one, unless you want to do that one too?

Reply:
That definitely looks a lot better :)